visualization/src/resource-visualization.cc

```diff
@@ -304,7 +327,17 @@ void visualizeReprojectedDepthResource(
 
   // Publish accumulated point cloud in global frame.
   sensor_msgs::PointCloud2 ros_point_cloud_G;
-  backend::convertPointCloudType(accumulated_point_cloud_G, &ros_point_cloud_G);
+  if (FLAGS_vis_pointcloud_filter_dense_map_before_publishing) {
+    pcl::PointCloud<pcl::PointXYZI>::Ptr pcl_cloud(
+        new pcl::PointCloud<pcl::PointXYZI>);
+    CHECK_NOTNULL(pcl_cloud);
+    backend::convertPointCloudType(accumulated_point_cloud_G, &(*pcl_cloud));
+    voxelGridPointCloud(FLAGS_vis_pointcloud_filter_leaf_size_m, pcl_cloud);
+    backend::convertPointCloudType((*pcl_cloud), &ros_point_cloud_G);
+  } else {
+    backend::convertPointCloudType(
+        accumulated_point_cloud_G, &ros_point_cloud_G);
+  }
```

Comment:
Aren't you doing the same filter twice here (also in line 294-306?), once during the accumulation and once after the accumulation? Does that make sense?

Reply:
Let me double-check it was late

Reply:
The line 294 concerns the unaccumulated publisher and from line 329 we would filter the accumulated one. I'll make a tiny fix such that the local tf publisher also benefits from this filtering process, but for the accumulated one we should wait for all the points being added to the global struct.